File: Branch-SDK/NativeComputeDebugCell.m
//
//  NativeComputeDebugCell.m
//  Branch
//
//  Created by Usman Shafique on 8/26/22.
//  Copyright © 2022 Branch, Inc. All rights reserved.
//

#import "NativeComputeDebugCell.h"

@interface NativeComputeDebugCell()
@end

@implementation NativeComputeDebugCell

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    [self initialize];
    return self;
}

- (id)initWithCoder:(NSCoder *)coder {
    self = [super initWithCoder:coder];
    [self initialize];
    return self;
}

- (void)initialize {
    UITextView *textView = [[UITextView alloc] initWithFrame:CGRectZero];
    [textView setTranslatesAutoresizingMaskIntoConstraints:NO];
    [textView setEditable:NO];
    [textView setScrollEnabled:NO];
    [textView setEditable:NO];
    [textView setTextAlignment:NSTextAlignmentLeft];
    [textView setBackgroundColor:UIColor.lightGrayColor];
    [textView setTextColor:UIColor.blackColor];
    [self.contentView addSubview:textView];
    self.textView = textView;
    
    
    [NSLayoutConstraint activateConstraints:@[
       [self.textView.topAnchor constraintEqualToAnchor:self.contentView.topAnchor],
       [self.textView.leadingAnchor constraintEqualToAnchor:self.contentView.leadingAnchor],
       [self.textView.trailingAnchor constraintEqualToAnchor:self.contentView.trailingAnchor],
       [self.textView.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor]
    ]];
}

- (void)prepareForReuse {
    [super prepareForReuse];
    self.textView.text = nil;
}


@end

File: Branch-SDK/NativeComputeDebugCell.swift
import UIKit


@objc class NativeComputeDebugCell: UITableViewCell {

    weak var textView: UITextView!
        
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        self.initialize()
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.initialize()
    }

    func initialize() {
        // textview config
        let textView = UITextView(frame: .zero)
        textView.translatesAutoresizingMaskIntoConstraints = false
        textView.isEditable = false
        textView.isScrollEnabled = false
        textView.textAlignment = .left
        textView.backgroundColor = .lightGray
        textView.textColor = .black
        self.contentView.addSubview(textView)
        self.textView = textView
        
        
        NSLayoutConstraint.activate([
            self.textView.topAnchor.constraint(equalTo: self.contentView.topAnchor),
            self.textView.leadingAnchor.constraint(equalTo: self.contentView.leadingAnchor),
            self.textView.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor),
            self.textView.bottomAnchor.constraint(equalTo: self.contentView.bottomAnchor)
        ])
    }

    override func prepareForReuse() {
        super.prepareForReuse()
        self.textView.text = nil
    }
}
